### PR TITLE
BAEL_3301_testing  @configuration properties

### PR DESCRIPTION
**Pull request by M-Abdelbaset**

### CODE REVIEW

File: spring-boot-modules/spring-boot-testing/src/main/java/com/baeldung/boot/configurationproperties/CustomCredentialsConverter.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConfigurationPropertiesBinding
+public class CustomCredentialsConverter implements Converter<String, Credentials> {
+
+	@Override
+	public Credentials convert(String source) {
+		String data[] = source.split(",");
```
Comment (JonCook) [inline spring-boot-modules/spring-boot-testing/src/main/java/com/baeldung/boot/configurationproperties/CustomCredentialsConverter.java]: Typically we would put Array designators "[]" on the type, not the variable.

File: spring-boot-modules/spring-boot-testing/src/test/java/com/baeldung/boot/configurationproperties/BindingPropertiesToBeanMethodsUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+@ContextConfiguration(classes = ServerConfigFactory.class)
+@TestPropertySource("classpath:server-config-test.properties")
+public class BindingPropertiesToBeanMethodsUnitTest {
+
+	@Autowired
+	private ServerConfigFactory configFactory;
+
+	@Test
+	void givenBeanAnnotatedMethod_whenBindingProperties_thenAllFieldsAreSet() {
+
```
Comment (JonCook) [inline spring-boot-modules/spring-boot-testing/src/test/java/com/baeldung/boot/configurationproperties/BindingPropertiesToBeanMethodsUnitTest.java]: Let's remove these empty lines at the start of each test method.

File: spring-boot-modules/spring-boot-testing/src/main/java/com/baeldung/boot/configurationproperties/CustomCredentialsConverter.java (hunk 1 above)
Comment (M-Abdelbaset) [reply spring-boot-modules/spring-boot-testing/src/main/java/com/baeldung/boot/configurationproperties/CustomCredentialsConverter.java]: Fixed.

File: spring-boot-modules/spring-boot-testing/src/test/java/com/baeldung/boot/configurationproperties/BindingPropertiesToBeanMethodsUnitTest.java (hunk 2 above)
Comment (M-Abdelbaset) [reply spring-boot-modules/spring-boot-testing/src/test/java/com/baeldung/boot/configurationproperties/BindingPropertiesToBeanMethodsUnitTest.java]: Done

**Pull request merged**